sky/unit/test/engine/mock_events.dart

```diff
@@ -79,3 +81,71 @@ class TestGestureEvent extends sky.GestureEvent {
   double velocityX;
   double velocityY;
 }
+
+class _PointerState {
+  bool isDown = false;
+  sky.Point location;
+}
+
+class TestPointerGenerator {
```

Comment:
instead of a TestPointerGenerator and ints used to identify a pointer, how about a TestPointer that automatically assigns itself an ID, and has all the methods below but applies them to itself directly?

As in:

```dart
class TestPointer {
  TestPointer() {
    _id = _nextId;
    _nextId += 1;
  }

  static int _nextId = 1;

  int _id;
  int get id => _id;
  bool _isDown = false;
  sky.point _location;

  sky.PointerEvent down([ sky.Point location = sky.Point.origin ]) {
    assert(!_isDown);
    _isDown = true;
    _location = location; // (or maybe use last location?)
    return new TestPointerEvent(
      type: 'pointerdown',
      pointer: _id,
      x: location.x,
      y: location.y
    );
  }

  // etc...
}
```